---
title: "Week 10 Questions"

format:
  html:
    code-fold: false
    code-tools: true
    code-copy: true
    highlight-style: github
    code-overflow: wrap
---

1.  **Looking back on the quarter what are two things you were not expecting to learn but did, glad you did, and how would you implement this knowledge beyond this class?**  I was not expecting to learn Quarto and clarifying the modes of DNA methylation via CG motifs. I look forward to exploring them more in my own work as a potential pattern to various toxin exposure response.

2.  **Rank the following from 1-5, with 1 being your best friend and 5 being your enemy (raven, rstudio, knitr, bash, github)**  
Raven ranks the highest at a 1 - gotta love a readily available coding platform that has enough juice to get your assignments finished almost in time. Bash ranks next at a 2 because it is exactly what it looks like at all times - no updates or crowd-sourced packages to thwart progress. Rstudio/ R is coming in at the most neutral 3 of all; for all the good it does, it does just as much evil. Knitr and github are tied for 4th and 5th for my enemies - knotr nevers works on the first run and github likes to see me struggle.

3.  **As part of your research project you must of learned something beyond methods. What is the most important biological finding from your work to to date?** Assuming you follow this line of research what is your long term goal of what you expect to learn about biology, ecology, or the environment?  So far, I've learned that there is a clustering of gene expression differences in both the synthetic horomone exposure as well as in the biotoxin exposure that is comparable to the expression in response to hypoxia. The genes are all upregulated in the face of these toxins, so I am looking forward to doing further work to tease apart separate or common differentially expressed genes across treatments.
